### PR TITLE
Implement authentication behavior on frontend

### PR DESCRIPTION
**Pull request by jankrokos**

### CODE REVIEW

File: rms_fe/src/clients/auth.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { LoginCredentials } from "../types/api/auth"
+import axios from "axios"
+
+axios.defaults.withCredentials = true
```
Comment (CreativePhilip) [inline rms_fe/src/clients/auth.ts]: This is present in the `setupAxios` function so this is unnecessary and can be removed

File: rms_fe/src/router.tsx [hunk 2]
```diff
@@ -1,10 +1,37 @@
...
+        isLoggedIn().then((loggedIn) => {
+            if (!loggedIn) {
+                navigate("/login")
+            }
+            setIsLoading(false)
+        })
+    }, [])
+
+    if (isLoading) {
+        return <></>
```
Comment (CreativePhilip) [inline rms_fe/src/router.tsx:30]: You should check if user is null here, and this makes you don't need the isLoading state

**Review by jsyska — APPROVED** [review]

File: rms_fe/src/router.tsx (hunk 2 above)
Comment (jankrokos) [reply rms_fe/src/router.tsx:30]: fixed

File: rms_fe/src/clients/auth.ts (hunk 1 above)
Comment (jankrokos) [reply rms_fe/src/clients/auth.ts]: fixed

**Review by CreativePhilip — APPROVED** [review]

**Pull request merged**